helpers/auth/error_handler.py

```diff
@@ -27,3 +28,33 @@ def __enter__(self):
 
     def __exit__(self, type, value, traceback):
         return False
+
+
+class CheckDuplicationManager():
```

Comment:
Great work here! Is it possible for you to do these checks for name and state in the SaveContextManager class above? This could save you the time of writing a new entire class to do the same.

Reply:
Thanks, @Bonifase,  it has not been possible so far to have both managers combined.
The `SaveContextManager` checks for an error message from the database. It does not consider the state of the data being sent.

The `CheckDuplicationManager` on the other hand checks for the name and the state without using the error messages from the database.

So the logic used in the two managers is different and combining them has been unsuccessful as we had seen in the sync I had earlier with you.